src/Sarif.Multitool/Rules/SARIF2010.ProvideCodeSnippets.cs

```diff
@@ -65,5 +78,46 @@ private void AnalyzeRegion(Region region, string regionPointer)
                     nameof(RuleResources.SARIF2010_ProvideCodeSnippets_Note_Default_Text));
             }
         }
+
+        private bool AnalyzeArtifactLocation(ArtifactLocation artifactLocation)
+        {
+            // no artifactLocation / no artifacts, so we should look for the snippet
+            if (artifactLocation == null || this.artifacts == null)
+            {
+                return true;
+            }
+
+            artifactLocation.TryReconstructAbsoluteUri(this.originalUriBaseIds, out Uri resolvedUri);
+
+            foreach (Artifact artifact in this.artifacts)
+            {
+                // content/text doesn't exist, continue to next
+                if (string.IsNullOrEmpty(artifact.Contents?.Text))
+                {
+                    continue;
+                }
+
+                artifact.Location.TryReconstructAbsoluteUri(this.originalUriBaseIds, out Uri artifactUri);
```

Comment:
>TryReconstructAbsoluteUri [](start = 34, length = 25)

```
if (!artifact.Location.TryReconstructAbsoluteUri(this.originalUriBaseIds, out Uri artifactUri)
{
    continue;
}
```
Need a test case for this too (an artifact that _does_ have content but that _cannot_ be resolved.

Reply:
Same thing as before, what if we don't have the uribaseid but the path is right, so we have to continue validating. Right?

---
In reply to: [462543051](https://github.com/microsoft/sarif-sdk/pull/2019#discussion_r462543051) [](ancestors = 462543051)

Reply:
You are right. Please do add the test case.

---
In reply to: [462547813](https://github.com/microsoft/sarif-sdk/pull/2019#discussion_r462547813) [](ancestors = 462547813,462543051)

Reply:
I added some comments in the sarif, so we can understand what i'm validating. Let me know what do you think

---
In reply to: [462576825](https://github.com/microsoft/sarif-sdk/pull/2019#discussion_r462576825) [](ancestors = 462576825,462547813,462543051)